src/services/FacilityService.ts

```diff
@@ -601,6 +602,58 @@ const fetchFacilityCountByGroup = async (facilityGroupIds: any): Promise<any> =>
   }, {})
 }
 
+const fetchProductStoreCountByGroup = async (facilityGroupIds: any): Promise<any> => {
+  if (!facilityGroupIds.length) return []
+  const requests = []
+
+  const facilityGroupIdList = facilityGroupIds
+  while (facilityGroupIdList.length) {
+    const batch = facilityGroupIdList.splice(0, 10)
+    const params = {
+      inputFields: {
+        facilityGroupId: batch,
+        facilityGroupId_op: "in"
+      },
+      viewSize: 250, // maximum view size
+      entityName: 'ProductStoreFacilityGroup',
+      noConditionFind: "Y",
+      filterByDate: 'Y',
+      fieldList: ['facilityGroupId', 'productStoreId']
+    }
+    requests.push(params)
+  }
+
+  const productStoreCountResponse = await Promise.allSettled(requests.map((params) => api({
+    url: 'performFind',
+    method: 'POST',
+    data: params
+  })))
+
+  const hasFailedResponse = productStoreCountResponse.some((response: any) => hasError(response.value) && !response?.data?.count)
+
+  if (hasFailedResponse) {
+    logger.error('Failed to fetch product store count for some groups')
+  }
+
+  // taking out the response from Promise.allSettled's 'value' field first 
+  const allResponseData = productStoreCountResponse.map((response: any) => response.value)
+    .reduce((responseData: any, response: any) => {
+      if (!hasError(response)) {
+        responseData.push(...response.data.docs)
+      }
+      return responseData
+    }, [])
+
+  return allResponseData.reduce((productStoreCountByGroup: any, responseData: any) => {
```

Comment:
Handle the case when the count for some facilityGroup is not found or the api fails. in this case we need to display 0 as count on the UI.